### DISCUSSION TITLE
posts/mesh

### DISCUSSION THREAD

**Discussion by giscus** [whexy/whexy-blog-comments#16] [General]
# posts/mesh

家用带宽为何抖动频繁？楼层网络为何缓慢不堪？千兆升级为何频频受阻？面对装修工人、电信师傅、电器商人的重重磨难，小石能否顺利完成家庭网络的升级改造工程？敬请收看大型流水账《十一天组网日记》。

https://www.whexy.com/posts/mesh

**Comment by Tonny-Gu** [comment]
超五类线可以跑到1Gbps，按照规范应该是100m下跑2.5G，[Ref](https://zhuanlan.zhihu.com/p/136479005)。上tb买个网线钳就可以自己压水晶头了。自己压不难，不需要剥线皮。

**Comment by Tert-butyllithium** [comment]
我们家当时装宽带也经历了重重困难（比如，装修时没有预留光纤入户），但是整体配置更简单。大概是两点原因：
1. 我家房子小，一个路由器就可以全覆盖🤣
2. 我家装的电信宽带，况且速度是100m

**Comment by whexy** [comment]
后续更新：
1. 中国移动宽带安装人员表示“光猫不能改桥接”，纯属谎言。
2. 文章中因中国移动修改了超级密码而无法使用有线回程。此超级密码可以通过社会工程学方式得到。（例如，给宽带安装人员塞一包烟、闲鱼花 5 块钱找内鬼要密码等等）

**Comment by ziqin** [comment]
> DHCPv6 是 IPv6 版本的 DHCP，负责分发内网 IPv6 地址，地址以 FC 或 FD 开头。但这个协议遭到了广泛批评，因为它破坏了 IPv6 的公共属性，违背了 IPv6 消灭 NAT 的愿景。由于谷歌的强硬态度，Android 系统直到现在（Android 12）也没有支持 DHCPv6。家用网络的配置需要避免 DHCPv6 的使用。
>
> 然而，公网 IPv6 分配无形中增大了攻击面。由于每个设备都有公网地址，针对隐藏端口的扫描和攻击会更加容易。所以工业界大规模使用 DHCPv6。

上述说法有些错误。

* DHCPv6 并非只能分发前缀为 `fc00::/7` 的“内网”地址，它亦可分发 Global Unicast Address。
* 用 DHCPv6 并不意味着要一定要引入 NAT (NAPT)，协议本身并未“破坏” IPv6 的 global addressability。
* 使用 DHCPv6 并非为了减小攻击面，相较 SLAAC 而言亦不能减小攻击面。正经的企业网络应配置好防火墙规则。

Android 之所以不支持 DHCPv6，主要是考虑到“用 Android 设备（经USB连接）给下级设备作热点”这种场景下，若该 Android 设备自身所接入的上级网络只允许它这条链路拿到单个 IPv6 GUA，则其下级设备无法获得 IPv6 GUA。而实际上，只要配置得当，DHCPv6 也可支持多个 GUA 地址；只不过实践中很多人部署用 DHCPv6 分配 IPv6 地址的网络时，延续了 IPv4 的习惯，配置成只允许一两个 IPv6 地址；这种情况下，若不做一对多的地址翻译，下级设备就没网。Google 拒绝 DHCPv6 进入 AOSP 的做法实际上是利用自身影响力，要挟要配置 IPv6 的网络管理员：必须支持 SLAAC。然而，DHCPv6 毕竟是被广泛接纳的 IETF 标准，国内安卓手机厂商为了更完整的 IPv6 支持，又常常自己实现一遍 DHCPv6，造成了一卓各表的局面。😂

大型网络使用 DHCPv6 有多种原因，包括但不限于地址规划、兼容不支持 [RDNSS](https://www.rfc-editor.org/rfc/rfc8106.html) 的 Windows 8 等老系统、路径依赖等。

我配置自己家网络时也不开 DHCPv6，因为没有必要，SLAAC + RDNSS 足矣。上面说这么多，是不希望 DHCPv6 背黑锅。

**Reply by whexy** [reply]
学到了🙏
